tests/system/python/packages/test_fog.py

```diff
@@ -0,0 +1,574 @@
+        put_url = "/foglamp/category/deliveryPositive%20Sine"
+        make_put_connection(foglamp_url,put_url,data)
+        
+        time.sleep(wait_time)
+        
+        for LOOP in range(retries):
+            print (retries)
+            print (LOOP)
+            get_url = "/foglamp/asset/positive_sine?seconds=600"
+            data = make_get_connection(foglamp_url,get_url)                     
+            if "event" in data[0]["reading"] and "rule" in data[0]["reading"]:              
+                assert data[0]["reading"]["event"] == "triggered"
+                assert data[0]["reading"]["rule"] == "Positive Sine"
+                break
+            elif LOOP < retries - 1 :
+                #print ("Continue")
+                continue
+            else :
+                assert data[0]["reading"]["event"] != "","TIMEOUT! positive_sine event not fired."+ foglamp_url + "/foglamp/asset/positive_sine?seconds=600" 
+                assert data[0]["reading"]["rule"] != "","TIMEOUT! positive_sine event not fired."+ foglamp_url + "/foglamp/asset/positive_sine?seconds=600"
+                #print ("NotFound")
+              
+        print ("---- positive_sine event fired ----") 
+        
+        print ("======================= EVENT POSITIVE SINE COMPLETE =======================")
+        
+    def test_negative_sine_notification(self,foglamp_url,remove_data_file,retries,wait_time):
+        remove_data_file("/tmp/out")
+        
+            
+        data = {"name": "Negative Sine","description": "Negative Sine notification instance","rule": "Threshold","channel": "python35","notification_type": "retriggered","enabled": True}
+        post_url = "/foglamp/notification"
+        make_post_connection(foglamp_url,post_url,data); 
+        
+        # Upload Python Script (write_out.py)
+        url = foglamp_url + '/foglamp/category/deliveryNegative%20Sine/script/upload'
+        script_path = 'script=@scripts/write_out.py'
+        upload_script = "curl -sX POST '{}' -F '{}'".format(url,script_path)
+        exit_code = os.system(upload_script)
+        assert exit_code == 0  
+    
+           
+        data = {"asset": "sinusoid","datapoint": "sinusoid","condition": "<"}
+        put_url = "/foglamp/category/ruleNegative%20Sine"
+        make_put_connection(foglamp_url,put_url,data)
+        
+        
+        data = {"enable": "true"}
+        put_url = "/foglamp/category/deliveryNegative%20Sine"
+        make_put_connection(foglamp_url,put_url,data)
+        
+        for LOOP in range(retries):
+            if os.path.exists("/tmp/out"):
+                break
+            time.sleep(1)
+        
+        if LOOP == retries - 1 :
+            print ("TIMEOUT! negative_sine event not fired. No /tmp/out file.")
+        else :
+             print ("---- negative_sine event fired ----") 
+        
+        print ("======================= EVENT NEGATIVE SINE COMPLETE =======================")
+    
+class TestToggledEvent:    
```

Comment:
move TestToggledEvent (1) test and TestPositiveNegativeSineNotification (2) under TestEventEngine itself. and also add assertions as required.